### PR TITLE
Improve theming layout

### PR DESCRIPTION
**Pull request by skjnldsv**
- [x] Put the default system theme on its dedicated line
- [x] Make the images clickable (no keyboard, accessibility is handled by the radio)
- [x] Separate the layout for themes and fonts
Capture d’écran_2022-04-28_17-01-51

Ref https://github.com/nextcloud/server/pull/32195#issuecomment-1112056091

### CODE REVIEW

**Comment by artonge** [comment]
1. Shouldn't "System default theme" be a toggle and allow selecting a dark and light themes with a select box?
2. I don't see why we should hide "high contrast" theme behind a toggle, as proposed by Jan. "normal" and "high contrast" themes seem like different themes to me. So I would keep them separated, like it is right now.
3. Maybe move the dyslexia toggle at the top with "System default", so it is easier to find in the hypothetical case that there are a lot of themes.

**Comment by skjnldsv** [comment]
> * Shouldn't "System default theme" be a toggle and allow selecting a dark and light themes with a select box?

No, they are all exclusive. If you select system, it will fallback to whatever theme matches your OS settings. If you select any other theme, you will override your system settings. :)

> 2\. I don't see why we should hide "high contrast" theme behind a toggle, as proposed by Jan. "normal" and "high contrast" themes seem like different themes to me. So I would keep them separated, like it is right now.

Agreed! Let's see in the future


> 3\. Maybe move the dyslexia toggle at the top with "System default", so it is easier to find in the hypothetical case that there are a lot of themes.

Currently it's not opened for external Themes, but we have to think about it yes :)
I'd say it's good as is for now as it was like that on the old layout :wink:

**Comment by skjnldsv** [comment]
@szaimen I was waiting for reviews before compiling+merging :stuck_out_tongue_closed_eyes:

**Comment by szaimen** [comment]
> @szaimen I was waiting for reviews before compiling+merging 😝

I currently cannot review/test before compiling...

**Review by szaimen — APPROVED** [review]
lgtm and works

However this empty line here looks strange to me (not sure if we can work around that):
[image](https://user-images.githubusercontent.com/42591237/165817684-2f4b711f-2137-4dc5-ae5d-871f8477cb18.png)

and I cannot use the picture to disable dyslexic again:
[image](https://user-images.githubusercontent.com/42591237/165817800-898ab640-de9f-4a88-8021-8967db971902.png)


EDIT: resolved

**Comment by artonge** [comment]
> it will fallback to whatever theme matches your OS settings

Yes, but which theme will it choose? The default Light theme? What if I want Nextcloud to switch between High contrast themes? Or between two third party themes?

**Comment by skjnldsv** [comment]
> However this empty line here looks strange to me (not sure if we can work around that):

We need to update the dark theme description to reflect this is stable now.

File: apps/theming/src/components/ItemPreview.vue [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <template>
-	<div class="theming__preview">
-		<div class="theming__preview-image" :style="{ backgroundImage: 'url(' + img + ')' }" />
+	<div :class="'theming__preview--' + theme.id" class="theming__preview">
+		<div class="theming__preview-image" :style="{ backgroundImage: 'url(' + img + ')' }" @click="onToggle" />
```
Comment (artonge) [inline apps/theming/src/components/ItemPreview.vue:3]: Could we also listen for the click event on the title and description?

Comment (artonge) [reply apps/theming/src/components/ItemPreview.vue:3]: Also, clicking on the dyslexia img does not untoggle

Comment (skjnldsv) [reply apps/theming/src/components/ItemPreview.vue:3]: > Also, clicking on the dyslexia img does not untoggle

Fixed already

**Review by szaimen — APPROVED** [review]
lgtm and works

File: apps/theming/lib/Themes/DarkHighContrastTheme.php [hunk 2]
```diff
@@ -33,7 +33,7 @@ public function getId(): string {
 	}
 
 	public function getMediaQuery(): string {
-		return '(prefers-color-scheme: dark) and (prefers-contrast: more)';
+		return '(prefers-color-scheme: dark) and (prefers-contrast: high)';
```
Comment (nickvergessen) [inline apps/theming/lib/Themes/DarkHighContrastTheme.php:36]: https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-contrast#browser_compatibility
are the docs lying?

Comment (skjnldsv) [reply apps/theming/lib/Themes/DarkHighContrastTheme.php:36]: Wuuuut https://developer.mozilla.org/fr/docs/Web/CSS/@media/prefers-contrast

Comment (skjnldsv) [reply apps/theming/lib/Themes/DarkHighContrastTheme.php:36]: https://docs.w3cub.com/css/@media/prefers-contrast

Comment (skjnldsv) [reply apps/theming/lib/Themes/DarkHighContrastTheme.php]: Dropped the commit, it seems the french page is wrong, I opened a ticket
https://github.com/mdn/translated-content/issues/5364

**Comment by skjnldsv** [comment]
In the meantime, let get this merge shall we? :)

**Comment by szaimen** [comment]
> In the meantime, let get this merge shall we? :)

one approval missing 🤷‍♂️

**Review by nickvergessen — APPROVED** [review]

**Pull request merged**